model/task/task.go

```diff
@@ -1745,6 +1740,108 @@ func (t *Task) Archive() error {
 	return nil
 }
 
+func ArchiveMany(tasks []Task) error {
+	if len(tasks) == 0 {
+		return nil
+	}
+	// add execution tasks of display tasks passed in, if they are not there
+	execTaskMap := map[string]*Task{}
+	for _, t := range tasks {
+		execTaskMap[t.Id] = &t
+	}
+	additionalTasks := []string{}
+	for _, t := range tasks {
+		for _, et := range t.ExecutionTasks {
+			if execTaskMap[et] == nil {
+				additionalTasks = append(additionalTasks, t.ExecutionTasks...)
+				continue
+			}
+		}
+	}
+	if len(additionalTasks) > 0 {
+		toAdd, err := Find(ByIds((additionalTasks)))
+		if err != nil {
+			return errors.Wrap(err, "unable to find execution tasks")
+		}
+		tasks = append(tasks, toAdd...)
+	}
+
+	archived := []interface{}{}
+	taskIds := []string{}
+	for _, t := range tasks {
+		archived = append(archived, *t.makeArchivedtask())
+		taskIds = append(taskIds, t.Id)
+	}
+
+	mongoClient := evergreen.GetEnvironment().Client()
+	ctx, cancel := evergreen.GetEnvironment().Context()
+	defer cancel()
+	session, err := mongoClient.StartSession()
+	if err != nil {
+		return errors.Wrap(err, "unable to start session")
+	}
+	defer session.EndSession(ctx)
+
+	txFunc := func(sessCtx mongo.SessionContext) (interface{}, error) {
+		oldTaskColl := evergreen.GetEnvironment().DB().Collection(OldCollection)
```

Comment:
With the repotracker we do StartTransaction and then AbortTransaction in the cases of failure. Why do we not do that here? (Disclaimer: I don't know much about transactions)